src/backend/routers/agent.py

```diff
@@ -101,6 +112,31 @@ async def create_agent(
                 await update_or_create_tool_metadata(
                     created_agent, tool_metadata, session, ctx
                 )
+
+        # Consolidate agent files into one index in compass
+        if (
+            Settings().feature_flags.use_compass_file_storage
+            and created_agent.tools_metadata
+        ):
+            artifacts = next(
+                (
+                    tool_metadata.artifacts
+                    for tool_metadata in created_agent.tools_metadata
+                    if tool_metadata.tool_name == ToolName.Read_File
+                    or tool_metadata.tool_name == ToolName.Search_File
+                ),
+                [],
+            )
+            file_ids = list(
+                set(
+                    artifact.get("id")
+                    for artifact in artifacts
+                    if artifact.get("type") == "local_file"
```

Comment:
Are the artifact `type` values stored in some enum somewhere? I think ideally we compare to an enum. Not too familiar with what the possible values can be

Reply:
they're meant to be flexible for whatever tool. that being said i think a good idea is to just add it an enum class for artifacts in each tools file. ill go and do that. 👍